src/main/java/org/jenkinsci/plugins/ghprb/GhprbRepository.java

```diff
@@ -314,45 +310,43 @@ void onIssueCommentHook(IssueComment issueComment) throws IOException {
     }
 
     void onPullRequestHook(PullRequest pr) throws IOException {
+        GHPullRequest ghpr = pr.getPullRequest();
+        int number = pr.getNumber();
+        String action = pr.getAction();
 
-        ConcurrentMap<Integer, GhprbPullRequest> pulls = helper.getTrigger().getPulls();
+        ConcurrentMap<Integer, GhprbPullRequest> pulls = trigger.getPulls();
 
-        if ("closed".equals(pr.getAction())) {
-            pulls.remove(pr.getNumber());
-        } else if (helper.isProjectDisabled()) {
+        if ("closed".equals(action)) {
+            pulls.remove(number);
+        } else if (!trigger.isActive()) {
             logger.log(Level.FINE, "Not processing Pull request since the build is disabled");
-        } else if ("opened".equals(pr.getAction()) || "reopened".equals(pr.getAction())) {
-            GhprbPullRequest pull = pulls.get(pr.getNumber());
+        } else if ("opened".equals(action) || "reopened".equals(action)) {
+            GhprbPullRequest pull = pulls.get(number);
             if (pull == null) {
-                pulls.putIfAbsent(pr.getNumber(), new GhprbPullRequest(pr.getPullRequest(), helper, this));
-                pull = pulls.get(pr.getNumber());
+                pulls.putIfAbsent(number, new GhprbPullRequest(ghpr, trigger.getHelper(), this));
+                pull = pulls.get(number);
             }
-            pull.check(pr.getPullRequest());
-        } else if ("synchronize".equals(pr.getAction())) {
-            GhprbPullRequest pull = pulls.get(pr.getNumber());
+            pull.check(ghpr);
+        } else if ("synchronize".equals(action)) {
+            GhprbPullRequest pull = pulls.get(number);
             if (pull == null) {
-                pulls.putIfAbsent(pr.getNumber(), new GhprbPullRequest(pr.getPullRequest(), helper, this));
-                pull = pulls.get(pr.getNumber());
+                pulls.putIfAbsent(number, new GhprbPullRequest(ghpr, trigger.getHelper(), this));
+                pull = pulls.get(number);
             }
             if (pull == null) {
-                logger.log(Level.SEVERE, "Pull Request #{0} doesn''t exist", pr.getNumber());
+                logger.log(Level.SEVERE, "Pull Request #{0} doesn''t exist", number);
                 return;
             }
-            pull.check(pr.getPullRequest());
+            pull.check(ghpr);
         } else {
-            logger.log(Level.WARNING, "Unknown Pull Request hook action: {0}", pr.getAction());
+            logger.log(Level.WARNING, "Unknown Pull Request hook action: {0}", action);
         }
         GhprbTrigger.getDscp().save();
```

Comment:
If we don't end up doing anything with a comment or PR, do we need to save?  Looking at the thread dumps, we spend a LOT of time in this call (maybe synchronization when multiple jobs are doing this at the same time?)